### ISSUE TITLE
Can not parse an ActorPath: /user/Processing-actor

### ISSUE THREAD

**Issue by RFlipper** [akkadotnet/akka.net#3500]
I'm trying to use At Least Once Delivery pattern with Akka Remoting. 
It works well if I use IActorRef in the _Deliver_ method, but fails if I use ActorSelection. 

- Which Akka.Net version you are using
1.3.8
- On which platform you are using Akka.Net
.Net Framework  4.6.1
- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.

```
       public JobSenderActor(ActorSelection targetActor)
        {
            _targetActor = targetActor;

            Command<IBaseActorMessage>(command =>
            {
                Deliver(_targetActor, deliveryId => new ReliableDeliveryEnvelope<IBaseActorMessage>(command, deliveryId));

                SaveSnapshot(GetDeliverySnapshot());
            });
           ...

```

Stack trace: 
```
   at Akka.Actor.ActorPath.Parse(String path)
   at Akka.Persistence.AtLeastOnceDeliveryReceiveActor.Deliver(ActorSelection destination, Func`2 deliveryMessageMapper)
   at MyTest.API.JobSenderActor.<.ctor>b__6_2(IBaseActorMessage command)
   at lambda_method(Closure , Object , Action`1 , Action`1 , Action`1 , Action`1 , Action`1 )
   at Akka.Tools.MatchHandler.PartialHandlerArgumentsCapture`6.Handle(T value)
   at Akka.Persistence.ReceivePersistentActor.ExecutePartialMessageHandler(Object message, PartialAction`1 partialAction)
   at Akka.Persistence.ReceivePersistentActor.OnCommand(Object message)
   at Akka.Persistence.UntypedPersistentActor.ReceiveCommand(Object message)
   at Akka.Persistence.UntypedPersistentActor.Receive(Object message)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Persistence.Eventsourced.<ProcessingCommands>b__93_0(Receive receive, Object message)
```

Exception: 
`Can not parse an ActorPath: /user/Processing-actor
`

It looks like a bug because if I replace ActorSelection to 

```
    var recipient = _targetActor.ResolveOne(TimeSpan.FromSeconds(5)).Result;

    Deliver(recipient.Path, deliveryId => new ReliableDeliveryEnvelope<IBaseActorMessage>(command, deliveryId));

```

I think this is the issue, PathString doesn't contain information about Anchor 
[https://github.com/akkadotnet/akka.net/blob/3f924fda8d6ca55c5af90a710d23be8ce36d5c7a/src/core/Akka.Persistence/AtLeastOnceDeliveryReceiveActor.cs#L179
](https://github.com/akkadotnet/akka.net/blob/3f924fda8d6ca55c5af90a710d23be8ce36d5c7a/src/core/Akka.Persistence/AtLeastOnceDeliveryReceiveActor.cs#L179
)

**Comment by Rizzen** [comment]
Encountered same issue.
I believe that issue can be easily fixed with combination of Anchor Path and PathString. I'll provide PR with possible solution.

**Comment by Aaronontheweb** [comment]
Yeah, the idea behind this is that the delivery needs to be targeted at an `ActorPath` in order for paths to be persisted and recovered correctly in the future. `IActorRef`s can't be persisted and re-hydrated this way since each `IActorRef` is stamped with a unique ID that changes whenever an actor is stopped / restarted.

Using an `ActorSelection` instead of an `ActorPath` shouldn't be a big deal, IMHO, since it uses the same semantics.

**Comment by Rizzen** [comment]
Sure, but current behavior a little confusing, I believe it's better to fix it some way.

**Comment by Aaronontheweb** [comment]
@Rizzen I'm agreeing with you

**Comment by sean-gilliam** [comment]
Closed via #3606
